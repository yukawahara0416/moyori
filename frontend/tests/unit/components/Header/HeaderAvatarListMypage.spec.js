import { shallowMount } from '@vue/test-utils'
import Component from '@/components/Header/HeaderAvatarListMypage.vue'

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    currentUser: { data: { id: 1 } }
  }

  wrapper = shallowMount(Component, {
    propsData
  })
})

describe('props', () => {
  it('currentUser', () => {
    expect(wrapper.props().currentUser).toStrictEqual(propsData.currentUser)
    expect(wrapper.props().currentUser instanceof Object).toBe(true)
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
