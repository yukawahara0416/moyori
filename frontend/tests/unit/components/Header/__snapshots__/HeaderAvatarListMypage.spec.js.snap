// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-list-item-stub
  activeclass=""
  tag="div"
  to="[object Object]"
>
  <v-list-item-avatar-stub
    size="40"
  >
    <v-icon-stub
      btn=""
    >
      mdi-account-circle
    </v-icon-stub>
  </v-list-item-avatar-stub>
   
  <v-list-item-content-stub>
    <v-list-item-title-stub>
      マイページ
    </v-list-item-title-stub>
  </v-list-item-content-stub>
</v-list-item-stub>
`;
